config/common/config.go

```diff
@@ -85,3 +86,13 @@ func ExtractProjectID() reference.ExtractValueFn {
 		return strings.TrimPrefix(tr.GetID(), "projects/")
 	}
 }
+
+func ExtractFolderID() reference.ExtractValueFn {
+	return func(mr resource.Managed) string {
+		tr, ok := mr.(jresource.Terraformed)
+		if !ok {
+			return ""
+		}
+		return strings.TrimPrefix(tr.GetID(), "folders/")
+	}
+}
```

Comment:
The issue is that the API url is built incorrectly with the default reference. Instead of it being something like `/v2/folders/<folder id>` it becomes `/v2/folder/folders/<folder id>` which throws a 404 error.
Without function:
![Screenshot 2023-11-17 at 8 44 35 AM](https://github.com/upbound/provider-gcp/assets/79924581/dc7c7bb1-3ca3-4426-a109-7b8b0ec9d6b6)
With function:
![Screenshot 2023-11-17 at 8 49 33 AM](https://github.com/upbound/provider-gcp/assets/79924581/7ca59be9-8ea3-4290-b771-aa124452cbd6)